file_writer.go

```diff
@@ -187,6 +234,7 @@ func (f *FileWriter) Write(b []byte) (int, error) {
 	for off < len(b) {
 		n, err := f.blockWriter.Write(b[off:])
```

Comment:
Similarly, we could just do

```
if f.cipher != nil {
    n, err = cipher.StreamWriter{S: f.cipher, W: f.blockWriter}.Write(b[off:])
} else {
    n, err = f.blockWriter.Write(b[off:])
}
```

Reply:
It seems a bit more complicated in reality, but it's defenitely better than previous version.